test/e2e/framework/config/config.go

```diff
@@ -56,6 +62,14 @@ func (c *Config) Validate() error {
 	errs = append(errs, c.Addons.Validate()...)
 	errs = append(errs, c.Suite.Validate()...)
 
+	// Apply feature gate flag on the actual shared feature gate map using a
+	// pflag set with the copied value.
+	ps := pflag.NewFlagSet("", pflag.ContinueOnError)
+	utilfeature.DefaultMutableFeatureGate.AddFlag(ps)
+	if err := ps.Parse([]string{"--feature-gates=" + flag.CommandLine.Lookup("feature-gates").Value.String()}); err != nil {
+		errs = append(errs, fmt.Errorf("failed to parse --feature-gates flag: %w", err))
+	}
```

Comment:
So this takes the value of the `--feature-gates` flag on test binary that this PR defines and passes it as value to `--feature-gates` flag on the controller binary which is defined [here](https://github.com/jetstack/cert-manager/blob/master/pkg/util/feature/feature_gate.go#L28).
I was looking at [this comment](https://github.com/jetstack/cert-manager/blob/master/pkg/util/feature/feature_gate.go#L25-L27) and wondering whether it's relevant in this case, however here we actually want the flag value to persist for the whole runtime of this controller, so it's not the case that this is just for a duration of _one_ test and this approach makes sense 👍🏼